### PR TITLE
Historical voting

### PR DESCRIPTION
**Pull request by Mitchdev**
Shows vote UI when joining chat after the vote was initiated. #32 

May need more testing, but the from tests I have done it works perfectly.

### CODE REVIEW

**Comment by 11k** [comment]
There's a bug where if you connect to chat during the cooldown period (the 7 seconds after a poll ends or is stopped), the bars are stuck at zero/don't reflect the number of votes cast. Would you mind fixing this?

**Comment by Mitchdev** [comment]
This has now been fixed in my testing. @11k 
Casting vote now checks if it's in the timeframe of the voting.

File: assets/chat/js/vote.js [hunk 1]
```diff
@@ -93,6 +94,17 @@ class ChatVote {
     }
   }
 
+  canCastVote(timestamp) {
+    if (this.vote) {
+      return (
+        new Date(timestamp).getTime() -
+          (this.vote.start.getTime() + this.vote.time + VOTE_END_TIME) <
```
Comment (11k) [inline assets/chat/js/vote.js:101]:
```suggestion
          (this.vote.start.getTime() + this.vote.time) <
```

There's a bug where users can now cast votes during the cooldown. I think this change fixes it without breaking anything else, but I'd appreciate if you double-checked.

**Comment by Mitchdev** [comment]
Tested and works, Thanks @11k

**Comment by Mitchdev** [comment]
There is one small bug where casted votes for ended votes are posted to chat on refresh.
I can see if I can fix that. Not a big problem though.

**Comment by 11k** [comment]
I see what you mean. That bug always existed so I'm willing to let it slide. Can deal with it in a future PR if you'd like.

**Comment by Mitchdev** [comment]
Yep, will see if I can fix it in a future PR.

**Pull request merged**